drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1477,7 +1493,9 @@ static const struct eth_stm32_hal_dev_cfg eth0_config = {
 };
 
 BUILD_ASSERT(DT_ENUM_HAS_VALUE(MAC_NODE, phy_connection_type, mii) ||
-	     DT_ENUM_HAS_VALUE(MAC_NODE, phy_connection_type, rmii),
+	     DT_ENUM_HAS_VALUE(MAC_NODE, phy_connection_type, rmii) ||
+	     DT_ENUM_HAS_VALUE(MAC_NODE, phy_connection_type, rgmii) ||
```

Comment:
maybe use a ``IF_ENABLED(DT_HAS_COMPAT_STATUS_OKAY(st_stm32n6_ethernet),(check for rgmii and gmii))``, to only allow gigabit connection types on supported soc